document/src/main/java/com/yahoo/document/json/JsonReader.java

```diff
@@ -83,6 +84,35 @@ ParsedDocumentOperation readSingleDocument(DocumentOperationType operationType,
         return operation;
     }
 
+    /**
+     * Reads a single operation. The operation is not expected to be part of an array.
+     *
+     * @return the parsed document operation
+     */
+    public ParsedDocumentOperation readSingleDocument() {
```

Comment:
Please add unit test for the new function in document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java.